lib/FusionInventory/Agent/Task/Inventory/Generic/Ipmi/Fru.pm

```diff
@@ -0,0 +1,55 @@
+#
+# OcsInventory agent - IPMI lan channel report
+#
+# Copyright (c) 2008 Jean Parpaillon <jean.parpaillon@kerlabs.com>
+#
+# The Intelligent Platform Management Interface (IPMI) specification
+# defines a set of common interfaces to a computer system which system
+# administrators can use to monitor system health and manage the
+# system. The IPMI consists of a main controller called the Baseboard
+# Management Controller (BMC) and other satellite controllers.
+#
+package FusionInventory::Agent::Task::Inventory::Generic::Ipmi::Fru;
+
+use strict;
+use warnings;
+
+use FusionInventory::Agent::Tools;
+use FusionInventory::Agent::Tools::Generic;
+
+sub isEnabled {
+    my (%params) = @_;
+    return 0 if $params{no_category}->{psu};
+    return 1;
+}
+
+sub doInventory {
+    my (%params) = @_;
+
+    my $inventory = $params{inventory};
+    my $logger    = $params{logger};
+
+    my $fru = getIpmiFru(logger => $logger);
+    my $psu;
+
+    foreach my $descr (keys %$fru) {
+        next unless $descr =~ /^PS(\d+)/;
+
+        $psu = {
+            PARTNUM => $fru->{$descr}->{data}->{'Board Part Number'},
+            SERIAL  => $fru->{$descr}->{data}->{'Board Serial'},
```

Comment:
Are you sure this is the right serial number ? I tested on a system and found it also report a "Product Serial" but I also saw "Board Serial" and "Product Serial" in dmidecode output and found they weren't related to PowerSupply Serial Number.

Reply:
Please show me your output.